csharp/src/Ice/ConnectionI.cs

```diff
@@ -969,6 +969,12 @@ public override void message(ThreadPoolCurrent current)
                             }
                         }
 
+                        if (_maxDispatches > 0 && _dispatchCount == _maxDispatches)
```

Comment:
Should this be `_dispatchCount >= _maxDispatches`, since _dispatchCount can be incremented by more than 1 when we received a batch request?